music21/spanner.py

```diff
@@ -771,10 +811,12 @@ def getLast(self):
 
 
 # ------------------------------------------------------------------------------
-class _SpannerRef(t.TypedDict):
+class PendingAssignmentRef(t.TypedDict):
```

Comment:
Since this has been moved from a private to a public class it needs documentation.

Reply:
And see below on why offsetInScore should be optional, and staffKey might be not needed.

Reply:
Still need to document this.